### PR TITLE
Give parameters and DRTs

### PR DESCRIPTION
**Pull request by riteshsamala**
Please test by toggling switches in login.html. Please toggle them separately while you test each feature

### CODE REVIEW

File: www/js/tweakModel.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+						variable: this.authObj[prop].variable,
+						variableType: this.authObj[prop].variableType
+					}
+					this.updateAssistanceScore(curAuthID); 
+				}
+			}
+		},
+		*/
+
+		updateModel: function(showParams, showSchema){
```
Comment (jonwetzel) [inline www/js/tweakModel.js]: I like that the showSchema part can be easily changed to use different schema later if we want, that's good.

Show is confusing here, as it used for show/hide.  I think we should rename it to "give" everywhere, including the URL parameters (changed them from "ss/sp" to "gs/gp").


```suggestion
		updateModel: function(giveParams, giveSchema){
```

File: www/js/tweakModel.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			console.log("shows", showParams, showSchema);
```
Comment (jonwetzel) [inline www/js/tweakModel.js]:
```suggestion
			console.log("giving", giveParams, giveSchema);
```

File: www/js/tweakModel.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+				if(this.studObj.hasOwnProperty(prop) && this.studObj[prop].hasOwnProperty("ID")){
+					authStudIDMap[this.studObj[prop].authoredID] = this.studObj[prop].ID;
+				}
+			}
+			for(var prop in this.authObj){
+				//console.log("current prop is", prop, this.authObj.hasOwnProperty(prop));
+				if(this.authObj.hasOwnProperty(prop) && this.authObj[prop].hasOwnProperty("type") && this.authObj[prop].type == "quantity"){
+					var curAuthID = this.authObj[prop].ID;
+					var parentDet = this.findParentDetails(curAuthID);
+					var schemaCheck = (showSchema == "on" && showParams != "on") ? ( parentDet.schema == schema ) : (showParams == "on"? true: false);
```
Comment (jonwetzel) [inline www/js/tweakModel.js]:
```suggestion
					var schemaCheck = (giveSchema == "on" && giveParams != "on") ? ( parentDet.schema == schema ) : (giveParams == "on"? true: false);
```

File: www/js/tweakModel.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+							variable: this.authObj[prop].variable,
+							variableType: this.authObj[prop].variableType
+						}
+						authStudIDMap[curAuthID] = "id" + getNewID;
+						getNewID = getNewID + 1;
+						this.updateAssistanceScore(curAuthID);
+					} 
+				}
+			}
+			if(showSchema == "on"){
```
Comment (jonwetzel) [inline www/js/tweakModel.js]:
```suggestion
			if(giveSchema == "on"){
```

File: www/js/session.js [hunk 5]
```diff
@@ -53,6 +54,17 @@ define([
 				handleAs: "json"
 			}).then(lang.hitch(this, function(model_object){
 				console.log("model object received", model_object);
+				if(params.sp == "on" || params.ss == "on"){
```
Comment (jonwetzel) [inline www/js/session.js]:
```suggestion
				if(params.gp == "on" || params.gs == "on"){
```

File: www/js/session.js [hunk 6]
```diff
@@ -53,6 +54,17 @@ define([
...
+					//check for student 
+					var TM = new tweakModel(model_object);
+					var new_obj = TM.updateModel(params.sp, params.ss);
```
Comment (jonwetzel) [inline www/js/session.js]:
```suggestion
					var new_obj = TM.updateModel(params.gp, params.gs);
```

File: www/js/tweakModel.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+			}
+			//traverse through authObj
+			for(var prop in this.authObj){
+				//console.log("current prop is", prop, this.authObj.hasOwnProperty(prop));
+				if(this.authObj.hasOwnProperty(prop) && this.authObj[prop].hasOwnProperty("type") && this.authObj[prop].type == "quantity"){
+					var curAuthID = this.authObj[prop].ID;
+					var parentDet = this.findParentDetails(curAuthID);
+					var schemaCheck = (giveSchema == "on" && giveParams != "on") ? ( parentDet.schema == schema ) : (giveParams == "on"? true: false);
+					var paramCheck = (giveParams == "on" && giveSchema != "on")? (this.authObj[prop].variableType == "parameter") : true;
+					if(schemaCheck && paramCheck && !authStudIDMap[curAuthID]){
```
[the remark this answers is not in the document]
Comment (riteshsamala) [reply www/js/tweakModel.js]: Agreed and tested, Thanks !

**Pull request merged**